### PR TITLE
Fall detection docs

### PR DESCRIPTION
**Pull request by hanryxu**
## What does this PR include?
Add demo Jupyter notebook to unattended baggage example.
## User API changes
None
## How did I test the PR?
Jupyter notebook tested in Google Colab
## New dependencies included
None

### CODE REVIEW

File: examples/fall_detection/demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Fall detection\n",
+    "\n",
+    "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/uclasystem/VQPy/blob/main/examples/fall_detection/demo.ipynb)\n",
+    "\n",
+    "## Introduction\n",
+    "\n",
+    "In this example we adopt code from [Human Falling Detection and Tracking](https://github.com/GajuuzZ/Human-Falling-Detect-Tracks) to track human movement and detect action. Two models are used: [AlphaPose](https://github.com/MVIG-SJTU/AlphaPose), to get person's body keypoints from cropped image of person; and [ST-GCN](https://github.com/yysijie/st-gcn), to predict action from every 30 frames of each person's keypoints."
```
Comment (yushan111) [inline examples/fall_detection/demo.ipynb:14]: adopt code -> we use models in ``

Comment (yushan111) [reply examples/fall_detection/demo.ipynb:14]: Leave the details in `Step 1: Define VObj type for person`. Introduction should only include general information

File: examples/fall_detection/demo.ipynb [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# loading the two models for inference\n",
+    "pose_model = SPPE_FastPose('resnet50', 224, 160, device='cuda',\n",
+    "        weights_file=os.path.join(\n",
+    "            os.path.abspath(model_dir), \"fast_res50_256x192.pth\"\n",
+    "        )\n",
+    "    )\n",
+    "action_model = TSSTG(\n",
```
Comment (yushan111) [inline examples/fall_detection/demo.ipynb:139]: move them inside the person class

File: examples/fall_detection/demo.ipynb [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Adapt model from pose prediction\n",
+    "\n",
+    "Two models are used to predict the pose a person:\n",
+    "\n",
+    "- AlphaPose: takes the frame and person's bounding box, returns a list of keypoints. Keypoints are mid-products to be used in ST-GCN.\n",
```
Comment (yushan111) [inline examples/fall_detection/demo.ipynb:123]: add the reference here

File: examples/fall_detection/demo.ipynb [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class Person(vqpy.VObjBase):\n",
+    "    pass"
```
Comment (yushan111) [inline examples/fall_detection/demo.ipynb:111]: remove.

File: examples/fall_detection/demo.ipynb [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+    "        action_name = action_model.class_names[out[0].argmax()]\n",
+    "        return action_name"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> The conditional statements `if tlbr is None` in L11 and iteration over `for i in range(-self._track_length, 0)` in L21 are attempting to get the most recent bounding box and keypoints. We are still working on more convenient ways to access properties (and stateful properties) in VObjs."
```
Comment (yushan111) [inline examples/fall_detection/demo.ipynb:202]: no need to mention

File: examples/fall_detection/demo.ipynb [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2: Query on `Person`'s pose\n",
+    "\n",
+    "To filter on people that are falling down, we filter on `pose` having value `\"Fall Down\"` (7 actions should be supported: `\"Standing\", \"Walking\", \"Sitting\", \"Lying Down\", \"Stand up\", \"Sit down\", \"Fall Down\"`).\n",
```
Comment (yushan111) [inline examples/fall_detection/demo.ipynb:212]: > (7 actions should be supported: `\"Standing\", \"Walking\", \"Sitting\", \"Lying Down\", \"Stand up\", \"Sit down\", \"Fall Down\"`)
move to the explanation of `pose` property

File: examples/fall_detection/demo.ipynb (hunk 1 above)
Comment (hanryxu) [reply examples/fall_detection/demo.ipynb]: Moved away.

File: examples/fall_detection/demo.ipynb (hunk 3 above)
Comment (hanryxu) [reply examples/fall_detection/demo.ipynb]: Moved reference here.

File: examples/fall_detection/demo.ipynb (hunk 5 above)
Comment (hanryxu) [reply examples/fall_detection/demo.ipynb]: Removed

File: examples/fall_detection/demo.ipynb (hunk 6 above)
Comment (hanryxu) [reply examples/fall_detection/demo.ipynb:181]: Moved

File: examples/fall_detection/demo.ipynb (hunk 2 above)
Comment (hanryxu) [reply examples/fall_detection/demo.ipynb]: Moved

File: examples/fall_detection/demo.ipynb (hunk 4 above)
Comment (hanryxu) [reply examples/fall_detection/demo.ipynb]: Removed

File: examples/fall_detection/demo.ipynb [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Running the query\n",
+    "\n",
+    "With the `Person` VObj and the query defined, we can run the query, with:\n",
+    "\n",
+    "- `cls_name` is a tuple for mapping numerical outputs of object detector to str\n",
+    "\n",
+    "\te.g. `vqpy.COCO_CLASSES` here starts with `(\"person\", \"bicycle\", ...)`, meaning that we will map output `0` of object detector to COCO class `\"person\"`\n",
```
Comment (yushan111) [inline examples/fall_detection/demo.ipynb:258]: change the doc to that in `unattended baggage`

File: examples/fall_detection/demo.ipynb [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+    "!mv VQPy/vqpy ./\n",
+    "# install VQPy's dependencies\n",
+    "!pip3 install lap cython_bbox shapely\n",
+    "import numpy as np\n",
+    "import torch\n",
+    "import vqpy\n",
+    "import os\n",
+    "import sys\n",
+    "sys.path.append(\"VQPy/examples/fall_detection/detect\")\n",
+    "video_path = \"./fall.mp4\"\t# path to video\n",
```
[the remark this answers is not in the document]
Comment (yushan111) [reply examples/fall_detection/demo.ipynb:53]: You may also need to change other examples correspondingly.

File: examples/fall_detection/demo.ipynb [hunk 9]
```diff
@@ -0,0 +1,323 @@
...
+    "# download VQPy, move vqpy/ to root directory for import\n",
+    "!git clone https://github.com/uclasystem/VQPy.git\n",
+    "!mv VQPy/vqpy ./\n",
+    "# install VQPy's dependencies\n",
+    "!pip3 install lap cython_bbox shapely\n",
+    "import numpy as np\n",
+    "import torch\n",
+    "import vqpy\n",
+    "import os\n",
+    "import sys\n",
```
[the remark this answers is not in the document]
Comment (yushan111) [reply examples/fall_detection/demo.ipynb:51]: same as before

File: examples/fall_detection/demo.ipynb [hunk 10]
```diff
@@ -0,0 +1,323 @@
...
+    "\n",
+    "In this example we use models in [Human Falling Detection and Tracking](https://github.com/GajuuzZ/Human-Falling-Detect-Tracks) to track human movement and detect action."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Environment setup\n",
```
Comment (yushan111) [inline examples/fall_detection/demo.ipynb:22]: ##

File: examples/fall_detection/demo.ipynb [hunk 11]
```diff
@@ -0,0 +1,323 @@
...
+    "\n",
+    "Python3.8 is recommended to avoid compatibility issues when installing YOLOX.\n",
+    "\n",
+    "Please download video from [here](https://youtu.be/ctniCxIdpTY) and place it in the same directory as this notebook.\n",
+    "\n",
+    "You'll also need to download pre-trained models from [SPPE FastPose (AlphaPose)](https://drive.google.com/file/d/1IPfCDRwCmQDnQy94nT1V-_NVtTEi4VmU/view?usp=sharing) and [ST-GCN](https://drive.google.com/file/d/1mQQ4JHe58ylKbBqTjuKzpwN2nwKOWJ9u/view?usp=sharing) and place them in the same directory. (models from [Human Falling Detection and Tracking](https://github.com/GajuuzZ/Human-Falling-Detect-Tracks#pre-trained-models))"
```
[the remark this answers is not in the document]
Comment (hanryxu) [reply examples/fall_detection/demo.ipynb]: Added

File: examples/fall_detection/demo.ipynb (hunk 7 above)
Comment (hanryxu) [reply examples/fall_detection/demo.ipynb]: Changed

File: examples/fall_detection/demo.ipynb (hunk 8 above)
[the remark this answers is not in the document]
Comment (hanryxu) [reply examples/fall_detection/demo.ipynb]: Split

File: examples/fall_detection/demo.ipynb (hunk 10 above)
Comment (yushan111) [reply examples/fall_detection/demo.ipynb:22]: resolve this

**Review by shanyu-sys — APPROVED** [review]
LGTM

**Pull request merged**